arch/arm/core/cortex_m/Kconfig

```diff
@@ -70,6 +70,7 @@ config CPU_CORTEX_M7
 	# Omit prompt to signify "hidden" option
 	select ARMV7_M_ARMV8_M_MAINLINE
 	select ARMV7_M_ARMV8_M_FP if CPU_HAS_FPU
+	select ARCH_HAS_NOCACHE_MEMORY_SUPPORT if ARM_MPU
```

Comment:
Could you move this "ARCH_HAS..." option together with the reset of ARCH_HAS options in CPU_CORTEX_M?